sunscreen_compiler_common/src/context.rs

```diff
@@ -5,10 +5,73 @@ use petgraph::algo::is_isomorphic_matching;
 use petgraph::stable_graph::{NodeIndex, StableGraph};
 use petgraph::visit::{EdgeRef, IntoEdgeReferences, IntoNodeReferences};
 use petgraph::Graph;
+use radix_trie::Trie;
 use serde::{Deserialize, Serialize};
 
 use crate::{Operation, Render};
 
+/**
+ * Stores information about the nodes associated with a certain operation.
+ */
+#[cfg(feature = "debugger")]
+#[derive(Clone, Deserialize, Serialize, Debug)]
+pub struct Group {
+    /**
+     * The name of the `Group`, representing the name of the operation/gadget.
+     */
+    pub label: String,
+
+    /**
+     * A list of ID's for the nodes contained in the operation.
+     */
+    pub node_ids: Vec<u64>,
+}
+#[cfg(feature = "debugger")]
+impl Group {
+    /**
+     * Creates a new `Group` instance.
+     */
+    pub fn new(name: String) -> Self {
+        Group {
+            label: name.to_string(),
+            node_ids: Vec::new(),
+        }
+    }
+
+    /**
+     * Adds a node ID to the group.
+     */
+    pub fn add_node<O: Operation>(mut self, node: NodeInfo<O>) {
```

Comment:
Remove. We'll just dynamically query nodes with a given group_id and implement caching if this is too slow. I don't think it will be though.